core_v2/sources/domains.move

```diff
@@ -0,0 +1,1232 @@
+        account::create_signer_with_capability(&borrow_global<CollectionCapability>(@aptos_names_v2).capability)
+    }
+
+    public fun get_burn_signer_address(): address acquires CollectionCapability {
+        account::get_signer_capability_address(
+            &borrow_global<CollectionCapability>(@aptos_names_v2).burn_signer_capability
+        )
+    }
+
+    fun get_burn_signer(): signer acquires CollectionCapability {
+        account::create_signer_with_capability(
+            &borrow_global<CollectionCapability>(@aptos_names_v2).burn_signer_capability
+        )
+    }
+
+    inline fun get_token_addr_inline(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapability {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &get_collection_name(is_subdomain(subdomain_name)),
+            &token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    public fun get_token_addr(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): address acquires CollectionCapability {
+        token::create_token_address(
+            &get_token_signer_address(),
+            &get_collection_name(is_subdomain(subdomain_name)),
+            &token_helper::get_fully_qualified_domain_name(subdomain_name, domain_name),
+        )
+    }
+
+    public fun get_record_obj(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): Object<NameRecord> acquires CollectionCapability {
+        object::address_to_object(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &NameRecord acquires CollectionCapability, NameRecord {
+        borrow_global<NameRecord>(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun get_record_mut(
+        domain_name: String,
+        subdomain_name: Option<String>,
+    ): &mut NameRecord acquires CollectionCapability, NameRecord {
+        borrow_global_mut<NameRecord>(get_token_addr_inline(domain_name, subdomain_name))
+    }
+
+    inline fun extract_subdomain_name(record: &NameRecord): Option<String> {
+        if (option::is_some(&record.subdomain_ext)) {
+            let subdomain_ext = option::borrow(&record.subdomain_ext);
+            option::some(subdomain_ext.subdomain_name)
+        } else {
+            option::none<String>()
+        }
+    }
+
+    inline fun get_collection_name(is_subdomain: bool): String {
+        if (is_subdomain) {
+            config::subdomain_collection_name()
+        } else {
+            config::domain_collection_name()
+        }
+    }
+
+    fun validate_registration_duration(
+        registration_duration_secs: u64,
+    ) {
+        assert!(
+            registration_duration_secs % SECONDS_PER_YEAR == 0,
```

Comment:
Why even take registration_duration_secs as seconds then? Why not take the number of years and convert to seconds here?